### PR TITLE
Story2task1

### PR DESCRIPTION
**Pull request by Roomsycakes01**
Probably the final version of my code, this git repo is a mess. Added form functionality and doc blocks.

### CODE REVIEW

File: add_item_form.php [hunk 1]
```diff
@@ -0,0 +1,160 @@
+<?php
+
+$db = new PDO ('mysql:host=db;dbname=collection', 'root', 'password');
+
+$db -> setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$nameQuery = $db->prepare( 'SELECT `name` FROM `games`;');
```
Comment (jesshamby) [inline add_item_form.php]: Thought you were leaving yourself a note for later but this is the name 😆 Love it!

File: add_item_form.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+     * The constructor for the Game object, only needs a name and the db to search the db and create a new object, if the add section is set to true th constructor can add a new item to the db
+     *
+     * @param String $name
+     * @param PDO $db
+     * @param bool $add
+     * @param String $genre
+     * @param String $length
+     * @param String $price
+     */
+    public function __construct(String $name, PDO $db, Bool $add = false, String $genre = '', String $length ='', String $price ='')
```
Comment (jesshamby) [inline add_item_form.php]: Look at this type-hinting - ON IT!

File: add_item_form.php [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    /**
+     * Checks to see if the inputted name is in the db.
+     *
+     * @param String $name
+     * @param PDO $db
+     * @return bool
+     */
+    protected function nameSearch( String $name, PDO $db) : Bool
+    {
+        $nameQuery = $db->prepare( 'SELECT `name` FROM `games`;');
```
Comment (jesshamby) [inline add_item_form.php]: Didn't you do this query at the start? Can you add this query into a function at the start and feed the result into here?

File: add_item_form.php [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    }
+}
+
+/**
+ * generates a list of all the games and info about them in the db
+ *
+ * @param PDO $db
+ * @return array
+ */
+function gamesArrayGen(PDO $db) : Array {
```
Comment (jesshamby) [inline add_item_form.php]: Picky but make sure the { starts on a new line :)

File: add_item_form.php [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    $nameQuery = $db->prepare( 'SELECT `name` FROM `games`;');
```
Comment (jesshamby) [inline add_item_form.php]: Used this same query again

File: add_item_form.php [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+/**
+ * generates a comma seperated list in html of the inputted array
+ *
+ * @param array $bigArray
+ */
+function listGen(Array $bigArray)
+{
+    echo '<ul>';
```
Comment (jesshamby) [inline add_item_form.php]: Was told it's best practice not to use echo within functions - try using concatenation instead :) create an empty string and .= your html

File: collection.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE html>
+<html lang='en'>
+<head>
+    <meta charset='UTF-8'/>
+    <meta name='viewport' content='width=device-width, initial-scale=1.0'/>
+    <title>Title</title>
```
Comment (jesshamby) [inline collection.php]: Can give a title :)

File: collection.php [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    <link rel='stylesheet' href='normalize.css'/>
+</head>
+<body>
+    <form method='POST'>
+        <input name='name' type='text'/>
```
Comment (jesshamby) [inline collection.php]: Can you label the inputs so you know the user is filling in data in the right place?

**Review by jesshamby — COMMENTED** [review]
Looks so so good! Been really picky for this reason :) 

I'm not as knowledgeable on OOP so it might be that Ash finds more things but I'm impressed :)

File: add_item_form.php (hunk 3 above)
Comment (Roomsycakes01) [reply add_item_form.php]: changed the object to just pull form the full db queried at the start

File: add_item_form.php (hunk 4 above)
Comment (Roomsycakes01) [reply add_item_form.php]: Only in objects i think :)

**Pull request merged**